### ISSUE TITLE
Incorrect compilation of nested ternaries.

### ISSUE THREAD

**Issue by otonashixav** [solidjs/solid#712]
**Describe the bug**
The compiler compiles nested ternaries incorrectly, resulting in a function being returned from the props getter instead of the expected value.

**To Reproduce**
See reproduction below.

**Expected behavior**
`get value()` should return the value passed in.

**Reproduction**
[Playground](https://playground.solidjs.com/?hash=1920360561&version=1.2.2)

**Comment by ryansolid** [comment]
Yeah I see it. We only optimize to a certain depth and escaping that optimization there is a bug. Thanks.

**Comment by ryansolid** [comment]
This is fixed in babel-plugin-jsx-dom-expressions@0.30.6. It should get pulled in with a full re-install but you might need to blow away your lock files to get this version until the next Solid release.

**Comment by ryansolid** [comment]
It actually wasn't that. I use a technique I call reactive onioning to reduce re-execution of dynamic things. The DOM reconciler will automatically layer each nested function into its own effect. So like:

```js
condA ? (condB ? resA : resB) : resC
```
If condB changed it only re-runs the inner condition. If A does it does everything. The downside of that is to achieve it the inner condition needs to be in its own function. This is in addition of putting guards around condition statements to force them into booleans to prevent unnecessary re-execution.

Now with DOM children there is no issue since well no one can intercept them. But with components you can, which is what you are seeing. With components I shouldn't treat children or props differently so we should do a different transform. The downside is that when condB changes the whole expression will re-evaluate. But that is probably fine as it is typically the payload not the conditions that are expensive and you'd need to do it anyway. The one downside is previously only downstream guards would be created, whereas this approach means re-creating the guards around conditions on every condition change. It's worthwhile from my perspective.

So after I make this change generally DOM elements will behave the same but with Components `<Show>` will be slightly more granular in execution on nested conditionals like this. Given toggling this sort of stuff back and forth isn't the type of thing one would animate I think it's fine.

**Comment by otonashixav** [comment]
I'm not sure it's fixed, but not sure if I've done something wrong when doing the reinstall. Here's the relevant parts of what I have: 

```tsx
<Input
  errorText={
    !formState.password
      ? t.common.input.error.required()
      : formState.password.length < 6
      ? t.dialog.setup.input.password.error.tooShort()
      : ""
  }
/>
```
compiles to:

@0.30.5:
```ts
...
get errorText() {
  return _c$2()
    ? t.common.input.error.required()
    : (() => {
        const _c$3 = _$memo(() => formState.password.length < 6, true);

        return () =>
          _c$3() ? t.dialog.setup.input.password.error.tooShort() : "";
      })();
},
...
```
@0.30.6:
```ts
...
get errorText() {
  return _$memo(() => !!!formState.password, true)()
    ? t.common.input.error.required()
    : () =>
        _$memo(() => formState.password.length < 6, true)()
          ? t.dialog.setup.input.password.error.tooShort()
          : "";
},
...
```

**Comment by ryansolid** [comment]
Yeah I see sorry.. I inlined but it looks like an extra wrapper is still sneaking in. I've fixed this 0.30.7.

**Comment by otonashixav** [comment]
Thanks! Works now.
